### PR TITLE
Document dynamic topic names

### LINKED ISSUE #780: Document dynamic topic routing
Documentation only inlcudes example when a message is routed to a single topic based on the configured topic name for the channel.

But both `@Outgoing` annotated method and `Emitter` is capable of routing message to a topic dynamicaly by setting `OutgoinMetadata`.

Example of this is:
```java
OutgoingKafkaRecordMetadata<?> metadata = OutgoingKafkaRecordMetadata.builder()
        .withTopic("movies")
        .withKey(movie.getYear())
        .build();
emitter.send(Message.of(movie).addMetadata(metadata));
```

Documentation should be updated to cover this functionality.

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes #780

### CODE REVIEW

**Comment by loicmathieu** [comment]
/cc @kenfinnigan 
To avoid duplication between example maybe I should remove the override of the topic name from the previous example ?

**Comment by kenfinnigan** [comment]
Probably not a bad idea @loicmathieu 

What do you think @cescoffier?

File: documentation/src/main/doc/modules/amqp/pages/outbound.adoc [hunk 1]
```diff
@@ -65,7 +65,19 @@ For example, you can configure the address, subjects, properties:
 include::example$outbound/AmqpOutboundMetadataExample.java[tag=code]
 ----
 
-NOTE: To be able to override the address per message, the connector is using an _anonymous sender_.
+=== Dynamic topic names
+
+Sometimes it is desirable to dynamically select the destination of a message, in this case you should not configure the name of the topic inside your application configuration file,
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/amqp/pages/outbound.adoc:70]:
```suggestion
Sometimes it is desirable to select the destination of a message dynamically.
In this case, you should not configure the address inside your application configuration file, but instead, use the outbound metadata to set the address.```

File: documentation/src/main/doc/modules/amqp/pages/outbound.adoc [hunk 2]
```diff
@@ -65,7 +65,19 @@ For example, you can configure the address, subjects, properties:
...
+but instead use the outbound metadata to set the name of the topic.
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/amqp/pages/outbound.adoc:71]: To be removed, part of my suggestion above.

File: documentation/src/main/doc/modules/amqp/pages/outbound.adoc [hunk 3]
```diff
@@ -65,7 +65,19 @@ For example, you can configure the address, subjects, properties:
...
+
+For example, you can route to a dynamic topic based on the incoming message:
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/amqp/pages/outbound.adoc:73]:
```suggestion
For example, you can send to a dynamic address based on the incoming message:
```

File: documentation/src/main/doc/modules/amqp/examples/outbound/AmqpOutboundDynamicTopicExample.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import io.smallrye.reactive.messaging.amqp.OutgoingAmqpMetadata;
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+public class AmqpOutboundDynamicTopicExample {
+    public Message<Double> metadata(Message<Double> incoming) {
+
+        // tag::code[]
+        String topicName = selectTopicFromIncommingMessage(incoming);
+        OutgoingAmqpMetadata metadata = OutgoingAmqpMetadata.builder()
+            .withAddress(topicName)
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/amqp/examples/outbound/AmqpOutboundDynamicTopicExample.java:12]: don't use `topic` for AMQP. Use address instead.

File: documentation/src/main/doc/modules/kafka/pages/outbound.adoc [hunk 5]
```diff
@@ -79,6 +79,19 @@ For example, you can configure the topic directly in the message, or add Kafka h
 include::example$outbound/KafkaOutboundMetadataExample.java[tag=code]
 ----
 
+=== Dynamic topic names
+
+Sometimes it is desirable to dynamically select the destination of a message, in this case you should not configure the name of the topic inside your application configuration file,
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/kafka/pages/outbound.adoc:84]:
```suggestion
Sometimes it is desirable to select the destination of a message dynamically. In this case, you should not configure the topic inside your application configuration file, but instead, use the outbound metadata to set the name of the topic.```

File: documentation/src/main/doc/modules/kafka/pages/outbound.adoc [hunk 6]
```diff
@@ -79,6 +79,19 @@ For example, you can configure the topic directly in the message, or add Kafka h
...
+but instead use the outbound metadata to set the name of the topic.
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/kafka/pages/outbound.adoc:85]: to be removed, part of my suggestion above.

File: documentation/src/main/doc/modules/kafka/pages/outbound.adoc [hunk 7]
```diff
@@ -79,6 +79,19 @@ For example, you can configure the topic directly in the message, or add Kafka h
...
+
+For example, you can route to a dynamic topic based on the incoming message:
+
+[source, java]
+----
+include::example$outbound/AmqpOutboundDynamicTopicExample.java[tag=code]
+----
+
+NOTE: To be able to set the address per message, the connector is using an _anonymous sender_.
```
Comment (cescoffier) [inline documentation/src/main/doc/modules/kafka/pages/outbound.adoc:94]: Hum... there is no such a thing in the Kafka world :-) ... yet

**Review by cescoffier — CHANGES_REQUESTED** [review]

**Comment by loicmathieu** [comment]
@cescoffier I suggestions applied!

Do you think I should remove the override of the topic/address name from the previous example  to avoid duplication ?

**Comment by cescoffier** [comment]
@loicmathieu Oh damned! I missed that. Yes, would be better to avoid the overlap.

**Comment by loicmathieu** [comment]
@cescoffier @kenfinnigan I remove the topic/address override from the previous section. Good to go ?

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
All good for me! Thanks @loicmathieu !

**Review by kenfinnigan — APPROVED** [review]
Thanks @loicmathieu

**Pull request merged**